crates/tako/src/internal/server/core.rs

```diff
@@ -177,10 +189,29 @@ impl Core {
             .append(&mut sleeping_mn_tasks);
 
         let worker_id = worker.id;
+        if let Some(g) = self.worker_groups.get_mut(&worker.configuration.group) {
+            g.worker_ids.insert(worker_id);
+        } else {
+            let mut worker_ids = Set::new();
+            worker_ids.insert(worker_id);
+            self.worker_groups.insert(
+                worker.configuration.group.clone(),
+                WorkerGroup { worker_ids },
+            );
+        }
         self.workers.insert(worker_id, worker);
     }
 
     pub fn remove_worker(&mut self, worker_id: WorkerId) -> Worker {
+        let worker = self.workers.get_worker(worker_id);
+        let group = self
+            .worker_groups
+            .get_mut(&worker.configuration.group)
+            .unwrap();
+        assert!(group.worker_ids.remove(&worker_id));
```

Comment:
`group.remove_worker`. Otherwise there's not much point in creating a separate `WorkerGroup` struct :) Its attribute probably shouldn't be visible.